frontend/src/firebase/firebaseInteractor.ts

```diff
@@ -507,4 +507,31 @@ export default class FirebaseInteractor {
     await this.auth.signOut();
     this.currentUser = null;
   }
+
+  // Researcher dashboard functions
+
+  async getAllStudents(): Promise<User[]> {
+    let students = (
+      await this.db
+        .collection("users")
+        .where("accountType", "==", "STUDENT")
+        .get()
+    ).docs;
+
+    return students.map((student) => {
+      let studentData = student.data();
+      return {
+        id: studentData.id,
+        name: studentData.name,
+        accountType: studentData.accountType,
+        age: studentData.age,
+        sessionId: studentData.sessionId,
+        onAssessment: studentData.onAssessment,
+        currentInterventionOrAssessment:
+          studentData.currentInterventionOrAssessment,
+        daysActive: studentData.daysActive,
+        profileIcon: studentData.profileIcon,
+      };
+    });
```

Comment:
Do you mind explaining this a little more?

Reply:
Right now we have the same code in `getUser`(I think this one is the actual correct name) and now this function. Since they are the same, can we consolidate into one place (this will also fix the problems of you not having values for sessionId and such, since that code has default values in place)?

Reply:
makes sense, let me know if what I changed looks like what you had in mind!

Reply:
That looks perfect!